### ISSUE TITLE
Get the secretmanager secret Arn

### ISSUE THREAD

**Issue by aflinton** [aws/aws-cdk#7949]
I am looking to get the arn of a secret so I can allow infrastructure I am deploying to have the permissions to get that secret e.g.:

const secret = secretsmanager.Secret.fromSecretArn(stack, secretName, secretArn);
 secret.grantRead(CdkBuild);

In this case to allow a codebuild build to be able to read a secret without me having to pass it in a way which appears in the logs (e.g. echo to file or... etc)

I can get the arn by using the AWS SDK listSecrets call and then iterating until the secretName matches but that is an async call and the value is returned after the cdk has passed the point where it needs the value.

I can get the secret value itself from within cdk e.g. for a github token:

ghtokenSecret = cdk.SecretValue.secretsManager(secretName, {
    jsonField: ghtoken,
 });

But can see no method/way to get the secret arn such that I can pass the value to my codebuild infrastructure.

### Use Case

Given multiple accounts for multiple environments (dev, sandbox, integration etc) allowing the use of just the SecretName allows me to use the cdk script effectively unchanged vs if using the arn as I have to create the secret for that account (using the same secret name) and then find it's arn and then insert that arn into some config (e.g. cdk.json) for each branch etc. which is potentially a source of failure/error.

This allows the secret to stay secret as opposed to echo'ing the value which appears in the cloudformation logs etc.

### Proposed Solution

Basically I would like something like:

const secret = secretsmanager.Secret.fromSecretName(stack, secretName);

and then secret has a .arn property

Though if I could get the secret using the above I might not need the arn, however it would fit the pattern of other arn based infrastructure.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by eikeon** [comment]
It'd be great to have the above proposed Secret.fromSecretName for the "multiple accounts for multiple environments" use case -- as then references could be kept relative to an account.

**Comment by skinny85** [comment]
Related: #7444, https://github.com/aws/aws-cdk/issues/6889#issuecomment-604277642
